python_modules/libraries/dagster-dbt/dagster_dbt/cli/resources_v2.py

```diff
@@ -130,6 +131,46 @@ def get_asset_key_by_output_name(self, output_name: str) -> AssetKey:
         """Get a dbt node's default asset key by its Dagster output name."""
         return self.node_info_to_asset_key(self.get_node_info_by_output_name(output_name))
 
+    def get_asset_key_for_dbt_unique_id(self, unique_id: str) -> AssetKey:
+        node_info = self.node_info_by_dbt_unique_id.get(unique_id)
+
+        if not node_info:
+            raise DagsterInvalidInvocationError(
+                f"Could not find a dbt node with unique_id: {unique_id}. A unique ID consists of"
+                " the node type (model, source, seed, etc.), project name, and node name in a"
+                " dot-separated string. For example: model.my_project.my_model"
+            )
+
+        return self.node_info_to_asset_key(node_info)
+
+    def get_asset_keys_for_dbt_source(self, source_name: str) -> Sequence[AssetKey]:
```

Comment:
Probably want to return a mapping of `table_name: str -> asset_key: AssetKey`, just to give some clarity to the user about what is producing the `AssetKey`
```suggestion
    def get_asset_keys_for_dbt_source(self, source_name: str) -> Mapping[str, AssetKey]:
```

Reply:
good call